sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveSubqueries.scala

```diff
@@ -43,7 +43,7 @@ case class PlanAdaptiveSubqueries(
             }
           )
         }
-        InSubqueryExec(expr, subqueryMap(exprId.id), exprId)
+        InSubqueryExec(expr, subqueryMap(exprId.id), exprId, shouldBroadcast = true)
```

Comment:
Why should we broadcast here? It's DPP with AQE which is the default case, and it's weird that your PR doesn't optimize the default case.

Reply:
This seems not for DPP. DPP in logical plan will be `DynamicPruningSubquery`. We transform it to `DynamicPruningExpression(InSubqueryExec(...))`.

For example, you can change this line to `shouldBroadcast = false` and still pass DPP test suites.

That is why I updated the doc of `InSubqueryExec`. Previously it said it is for DPP only, but it is not.

Reply:
Ah I see, thanks for the explaination!